lib/protocol/message.go

```diff
@@ -148,6 +156,13 @@ type Device struct {
 	Options         []Option // max:64
 }
 
+type FileDownloadProgressUpdate struct {
```

Comment:
Right, slim sounds like a good reason to me.

I do think it'd be good for the Request and FileDownloadProgressUpdate message types use consistent concepts, which might mean indexes instead of offsets. But, I'm not planning to badger you beyond this message.